aderyn_core/src/detect/nc/empty_blocks.rs

```diff
@@ -0,0 +1,88 @@
+use std::{collections::BTreeMap, error::Error};
+
+use crate::{
+    ast::{FunctionKind, NodeID, NodeType},
+    capture,
+    context::{
+        browser::GetClosestParentOfTypeX,
+        workspace_context::{ASTNode, WorkspaceContext},
+    },
+    detect::detector::{IssueDetector, IssueDetectorNamePool, IssueSeverity},
+};
+use eyre::Result;
+
+#[derive(Default)]
+pub struct EmptyBlockDetector {
+    // Keys are source file name and line number
+    found_instances: BTreeMap<(String, usize, String), NodeID>,
+}
+
+impl IssueDetector for EmptyBlockDetector {
+    fn detect(&mut self, context: &WorkspaceContext) -> Result<bool, Box<dyn Error>> {
+        for empty_block in context.blocks().iter().filter(|b| b.statements.is_empty()) {
+            if let Some(ASTNode::FunctionDefinition(f)) =
+                &empty_block.closest_parent_of_type(context, NodeType::FunctionDefinition)
+            {
+                // It's okay to have empty block if it's a constructor, receive, or fallback
```

Comment:
@alexroan I fixed it by changing the above from `.parent(context)` to `.closest_parent_of_type(context, NodeType::FunctionDefinition)`

Reply:
We would probably want this to capture constructors with nested empty blocks. Does this do that?

Reply:
At the moment as we have it, we don't capture any functions here instead we capture the starting `{` of the empty blocks so long as they are not inside a special function like constructor, receiver, fallback

Reply:
Do you think we should change this to capture the functions ?

Reply:
I have changed it to capture the functions now : ). I have also, added some more of my own test cases to the `EmptyBlocks.sol`

Follows this convention : 

```rust
// Good -> don't capture
// Bad -> capture
```